engine/execution/state/state.go

```diff
@@ -283,24 +283,48 @@ func (s *state) SaveExecutionResults(
 		trace.EXEStateSaveExecutionResults)
 	defer span.End()
 
+	header := result.ExecutableBlock.Block.Header
+
+	//outside batch because it requires read access
+	err := s.UpdateHighestExecutedBlockIfHigher(childCtx, header)
+	if err != nil {
+		return fmt.Errorf("cannot update highest executed block: %w", err)
+	}
+	return nil
+}
+
+func (s *state) saveExecutionResults(
+	ctx context.Context,
+	result *execution.ComputationResult,
+) error {
 	header := result.ExecutableBlock.Block.Header
 	blockID := header.ID()
 
+	err := s.chunkDataPacks.Store(result.AllChunkDataPacks())
```

Comment:
nit: If we use the named return value for error (see previous comment), we don't need to create a new error variable here.

```suggestion
	err = s.chunkDataPacks.Store(result.AllChunkDataPacks())
```